app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt

```diff
@@ -0,0 +1,337 @@
+        tempZones.group { }
+    }
+
+    /**
+     * Accept a material request delivery / return
+     */
+    private val acceptMaterialDelivery = { request: MaterialRequest ->
+        requests.set(
+            requests.indexOfFirst { it.requestId == request.requestId }, when (request.status) {
+                ACCEPTED -> request.copy(status = DELIVERING, staffInChargeId = staffUserId)
+                RETURN_REQUESTED -> request.copy(status = RETURNING, staffInChargeId = staffUserId)
+                else -> request // should never happen
+            }, this
+        )
+        Unit
+    }
+
+    /**
+     * Accept a material request delivery / return
+     */
+    private val onMaterialDelivered = { request: MaterialRequest ->
+        requests.set(
+            requests.indexOfFirst { it.requestId == request.requestId }, when (request.status) {
+                DELIVERING -> request.copy(status = DELIVERED)
+                RETURNING -> {
+                    for(item in request.items){
+                        val oldItemIndex = items.indexOfFirst { it.first.itemId == item.key }
+                        items[oldItemIndex] = items[oldItemIndex].copy(third = items[oldItemIndex].third + item.value)
+
+                    }
+                    request.copy(status = RETURNED)
+                }
+                else -> request // should never happen
+            }, this
+        )
+
+
+        Unit
+    }
+
+    /**
+     * Cancel a material request delivery / return
+     */
+    private val cancelMaterialDelivery = { request: MaterialRequest ->
+        requests.set(
+            requests.indexOfFirst { it.requestId == request.requestId }, when (request.status) {
+                DELIVERING -> request.copy(status = ACCEPTED, staffInChargeId = null)
+                RETURNING -> request.copy(status = RETURN_REQUESTED, staffInChargeId = null)
+                else -> request // should never happen
+            }, this
+        )
+        Unit
+    }
+
+    override fun onItemSelected(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
+        currentStatus.postValue(fromOrdinal(p2)!!)
+    }
+
+    override fun onNothingSelected(p0: AdapterView<*>?) {
+
+    }
+
+    companion object {
+        enum class StaffRequestStatus() {
+            DELIVERY,
+            RETURN,
+            OTHERS;
+
+
+            fun getItemRequestStatuses(): MutableList<Long> {
+                return when (this) {
+                    DELIVERY -> listOf(ACCEPTED, DELIVERING, DELIVERED)
+                    RETURN -> listOf(RETURN_REQUESTED, RETURNING, RETURNED)
+                    OTHERS -> listOf(PENDING, REFUSED)
+                }.map { it.ordinal.toLong() }.toMutableList()
+            }
+        }
+
+        /*
+                fun allStaffStatuses(): MutableList<Long> {
+                    return StaffRequestStatus.values().flatMap { it.getItemRequestStatuses() }
+                        .toMutableList()
+                }
+        */
```

Comment:
Is it useful to keep or can you remove it?

Reply:
Thanks, I forgot to remove it